beginner/apps/src/bin/radio-puzzle-3.rs

```diff
@@ -36,16 +37,18 @@ fn main() -> ! {
             // TODO insert the key-value pair
             // dict.insert(/* ? */, /* ? */).expect("dictionary full");
             } else {
-                log::error!("response packet was not a single byte");
+                defmt::error!("response packet was not a single byte");
                 dk::exit()
             }
         } else {
-            log::error!("no response or response packet was corrupted");
+            defmt::error!("no response or response packet was corrupted");
             dk::exit()
         }
     }
 
-    log::info!("{:?}", dict);
+    defmt::info!("{:?}", defmt::Debug2Format(&dict));
+    //                   ^^^^^^^^^^^^^^^^^^^ this adapter iscurrently needed to log `heapless`
```

Comment:
can't implement this for heapless due to circular dependency conundrum, see https://github.com/japaric/heapless/pull/172#issuecomment-741510859

Reply:
heapless 0.7.1 has defmt support (behind a Cargo feature); also const generics :tada: . we should move to that version ASAP

Reply:
nice! From what I see, it doesn't impl `Format` for `LinearMap`s though, right? PR incoming...